x-pack/plugins/observability_solution/synthetics/server/alert_rules/tls_rule/tls_rule.ts

```diff
@@ -107,45 +121,45 @@ export const registerSyntheticsTLSCheckRule = (
         }
 
         const alertId = cert.sha256;
-        const alertUuid = getAlertUuid(alertId);
-        const indexedStartedAt = getAlertStartedDate(alertId) ?? startedAt.toISOString();
-
-        const alertInstance = alertWithLifecycle({
+        const { uuid, start } = alertsClient.report({
           id: alertId,
-          fields: {
-            [CERT_COMMON_NAME]: cert.common_name,
-            [CERT_ISSUER_NAME]: cert.issuer,
-            [CERT_VALID_NOT_AFTER]: cert.not_after,
-            [CERT_VALID_NOT_BEFORE]: cert.not_before,
-            [CERT_HASH_SHA256]: cert.sha256,
-            [ALERT_UUID]: alertUuid,
-            [ALERT_REASON]: generateAlertMessage(TlsTranslations.defaultActionMessage, summary),
-          },
-        });
-
-        alertInstance.replaceState({
-          ...updateState(ruleState, foundCerts),
-          ...summary,
+          actionGroup: TLS_CERTIFICATE.id,
         });
-
-        alertInstance.scheduleActions(TLS_CERTIFICATE.id, {
+        const indexedStartedAt = start ?? startedAt.toISOString();
+
+        const payload = {
+          [CERT_COMMON_NAME]: cert.common_name,
+          [CERT_ISSUER_NAME]: cert.issuer,
+          [CERT_VALID_NOT_AFTER]: cert.not_after,
+          [CERT_VALID_NOT_BEFORE]: cert.not_before,
+          [CERT_HASH_SHA256]: cert.sha256,
+          [ALERT_UUID]: uuid,
+          [ALERT_REASON]: generateAlertMessage(TlsTranslations.defaultActionMessage, summary),
+        };
+
+        const context = {
           [ALERT_DETAILS_URL]: await getAlertUrl(
-            alertUuid,
+            uuid,
             spaceId,
             indexedStartedAt,
             alertsLocator,
             basePath.publicBaseUrl
           ),
+          ...updateState(ruleState, foundCerts),
```

Comment:
@ymao1 is the correct place for this?

Reply:
I assume Ying has more knowledge here, but this doesn't seem quite right,  especially since we're calling it again in the return of the executor.  And it doesn't look these fields were in the context before.

Reply:
Yea, I think you want to set the state during the initial reporting of the alert:

```
const { uuid, start } = alertsClient.report({
  id: alertId,
  actionGroup: TLS_CERTIFICATE.id,
  state: { ...updateState(ruleState, foundCerts), ...summary },
});
```

Reply:
okay thanks! I thought it was weird but wanted to make sure 🙂

Reply:
Resolved in this commit [1068617](https://github.com/elastic/kibana/pull/186551/commits/10686177bc6fc612581d951b54e62a696fd21c08)